src/array.jl

```diff
@@ -212,6 +212,31 @@ end
 
             convert($A{T, N, R}, A)
         end
+
+        function Base.vcat{T,N,R}(A1::$A{T, N, R}, An::$A...)
+            As = (A1, An...)
+            levels = unique(T[[a.pool.levels for a in As]...;])
```

Comment:
Haven't tested, but in general better work with `index` when dealing with `refs`, as they refer to it, not to `levels`. Worth having a test after reordering levels anyway.